x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_overview.tsx

```diff
@@ -17,69 +17,44 @@ import {
 import { SendAppSearchTelemetry as SendTelemetry } from '../../../shared/telemetry';
 import { FlashMessages } from '../../../shared/flash_messages';
-import { HttpLogic } from '../../../shared/http';
 import { LicensingLogic } from '../../../shared/licensing';
 
 import { EngineIcon } from './assets/engine_icon';
 import { MetaEngineIcon } from './assets/meta_engine_icon';
 import { ENGINES_TITLE, META_ENGINES_TITLE } from './constants';
 import { EnginesOverviewHeader, LoadingState, EmptyState } from './components';
 import { EnginesTable } from './engines_table';
+import { EnginesLogic } from './engines_logic';
 
 import './engines_overview.scss';
 
-interface GetEnginesParams {
-  type: string;
-  pageIndex: number;
-}
-interface SetEnginesCallbacks {
-  setResults: React.Dispatch<React.SetStateAction<never[]>>;
-  setResultsTotal: React.Dispatch<React.SetStateAction<number>>;
-}
-
 export const EnginesOverview: React.FC = () => {
-  const { http } = useValues(HttpLogic);
   const { hasPlatinumLicense } = useValues(LicensingLogic);
-
-  const [isLoading, setIsLoading] = useState(true);
-  const [engines, setEngines] = useState([]);
-  const [enginesPage, setEnginesPage] = useState(1);
-  const [enginesTotal, setEnginesTotal] = useState(0);
-  const [metaEngines, setMetaEngines] = useState([]);
-  const [metaEnginesPage, setMetaEnginesPage] = useState(1);
-  const [metaEnginesTotal, setMetaEnginesTotal] = useState(0);
-
-  const getEnginesData = async ({ type, pageIndex }: GetEnginesParams) => {
-    return await http.get('/api/app_search/engines', {
-      query: { type, pageIndex },
-    });
-  };
-  const setEnginesData = async (params: GetEnginesParams, callbacks: SetEnginesCallbacks) => {
-    const response = await getEnginesData(params);
-
-    callbacks.setResults(response.results);
-    callbacks.setResultsTotal(response.meta.page.total_results);
-
-    setIsLoading(false);
-  };
+  const {
+    dataLoading,
+    engines,
+    enginesTotal,
+    enginesPage,
+    metaEngines,
+    metaEnginesTotal,
+    metaEnginesPage,
+  } = useValues(EnginesLogic);
+  const {
+    fetchEngines,
+    fetchMetaEngines,
+    onEnginesPagination,
+    onMetaEnginesPagination,
+  } = useActions(EnginesLogic);
 
   useEffect(() => {
-    const params = { type: 'indexed', pageIndex: enginesPage };
-    const callbacks = { setResults: setEngines, setResultsTotal: setEnginesTotal };
-
-    setEnginesData(params, callbacks);
+    fetchEngines();
   }, [enginesPage]);
 
   useEffect(() => {
-    if (hasPlatinumLicense) {
-      const params = { type: 'meta', pageIndex: metaEnginesPage };
-      const callbacks = { setResults: setMetaEngines, setResultsTotal: setMetaEnginesTotal };
-
-      setEnginesData(params, callbacks);
-    }
+    if (hasPlatinumLicense) fetchMetaEngines();
   }, [hasPlatinumLicense, metaEnginesPage]);
 
-  if (isLoading) return <LoadingState />;
+  if (dataLoading) return <LoadingState />;
```

Comment:
We generally seem to use `dataLoading` over `isLoading` everywhere else in the repo so I changed this to match for consistency.